lib/cloudtasker/authenticator.rb

```diff
@@ -51,5 +58,19 @@ def verify(bearer_token)
     def verify!(bearer_token)
       verify(bearer_token) || raise(AuthenticationError)
     end
+
+    def oidc_token
+      google_metadata_server_url = 'http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/default/identity'
```

Comment:
Instead of manually retrieving an OIDC token from the metadata server, isn't there a prebuilt function in the GCP Ruby SDK to do so? (I haven't looked it up yet)

Reply:
From what I was able to research, I did not find a function in the Ruby SDK to fetch the token. Here are some relevant links that I used to try to find the answer:
* https://cloud.google.com/run/docs/authenticating/service-to-service#run-service-to-service-example-go
* https://cloud.google.com/ruby/docs/reference/google-cloud-binary_authorization/latest

In the first link, you can see that certain languages have a function to fetch the token and Ruby is not included.
The second links shows all the Ruby GCP libraries. I did not find anything relevent there unless by mistake I did not see it.

Because I couldn't find a library I decided to use manually retrieve the token as described at the end of the page in the first link.